example/RNWifiReborn/App.js

```diff
@@ -23,33 +24,68 @@ const App = () => {
   const [connected, setConnected] = useState({connected: false, ssid: 'S4N'});
   const [ssid, setSsid] = useState('');
   const password ="tanenbaum-1981";
-  const isWep = false;
 
-  const wifi = async () => {
+  const initWifi = async () => {
+    try {
+      const ssid = await WifiManager.getCurrentWifiSSID();
+      setSsid(ssid);
+      console.log('Your current connected wifi SSID is ' + ssid);
+    } catch (error) {
+      setSsid('Cannot get current SSID!' + error.message);
+      console.log('Cannot get current SSID!', {error});
+    }
+  }
+
+  const requestLocationPermission = async () => {
+    try {
+      const granted = await PermissionsAndroid.request(
+        PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
+        {
+          title: "React Native Wifi Reborn App Permission",
+          message:
+            "Location permission is required to connect with or scan for Wifi networks. ",
+          buttonNeutral: "Ask Me Later",
+          buttonNegative: "Cancel",
+          buttonPositive: "OK"
+        }
+      );
+      if (granted === PermissionsAndroid.RESULTS.GRANTED) {
+        initWifi();
+      } else {
+        console.log("Location permission denied");
+      }
+    } catch (err) {
+      console.warn(err);
+    }
+  };
+
+  const connectWithWifi = async () => {
     try {
       const data = await WifiManager.connectToProtectedSSID(
         ssid,
         password,
-        isWep,
+        false,
```

Comment:
I have applied this change but still not beneficial.It gets connected but throws error in callback

Reply:
Did you clean via gradle? (cd android && ./gradlew clean). What exception do you get?

Reply:
https://github.com/JuanSeBestia/react-native-wifi-reborn/issues/30

Reply:
In that case I assume you are using Android 7/8/9? What if you remove the wifi network first from Settings?